scripts/cangen/cangen/can_generator.py

```diff
@@ -247,13 +237,25 @@ def generate_code(bus: Bus, config: Config):
     logger.info("Code generation complete")
 
 
+def _prepare_output_directory(output_dir):
+    if os.path.exists(output_dir):
+        shutil.rmtree(output_dir)
+
+    os.makedirs(output_dir, exist_ok=True)
```

Comment:
these commands can be rearranged

1. make the folder, dont raise error if it exists
2. clear the folder with shutil, already know the folder exists since we just created it
3. create the gitignore

you don't need to check if the gitignore exists because we just cleaned the folder